capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -90,27 +102,91 @@
+    ArrayList<Card> studySetCards = new ArrayList<>();
+
+    try (Connection conn = pool.getConnection()) {
+      try (
+        PreparedStatement query_statement = conn.prepareStatement(
+          VIEW_STUDY_SET_QUERY
+        )
+      ) {
+        studySetID = studySetID.substring(1);
+        query_statement.setInt(1, Integer.parseInt(studySetID));
+        ResultSet result = query_statement.executeQuery();
+        while (result.next()) {
+          if (studySetDetails.containsKey("title") == false) {
+            String title = result.getString("title");
+            studySetDetails.put("title", title);
+
+            String description = result.getString("description");
+            studySetDetails.put("description", description);
+
+            String subject = result.getString("subject");
+            studySetDetails.put("subject", subject);
+
+            String user_author = result.getString("user_name");
+            studySetDetails.put("user_author", user_author);
+
+            String university = result.getString("name");
+            studySetDetails.put("university", university);
+          }
+          String cardFront = result.getString("front");
+          String cardBack = result.getString("back");
+          Card card = new Card(cardFront, cardBack);
+          studySetCards.add(card);
+        }
+        studySetDetails.put("cards", studySetCards);
+        return studySetDetails;
+      }
+    }
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
     ServletContext servletContext = getServletContext();
-    String search_word  = request.getParameter("stringToSearchBy");
     DataSource pool = (DataSource) servletContext.getAttribute("my-pool");
-    try {
-      ArrayList<HashMap<String, String>> studySets = runSqlQuery(
-        pool,
-        search_sql_statement,
-        search_word
-      );
-      response.setContentType("application/json;");
-      Gson gson = new Gson();
-      String studySetsResult = gson.toJson(studySets);
-      response.getWriter().println(studySetsResult);
-    } catch (SQLException ex) {
-      throw new RuntimeException(
-        "There is an error with your sql statement ... ",
-        ex
-      );
+    if (request.getPathInfo() == null) {
+      String search_word = request.getParameter("stringToSearchBy");
+      try {
+        ArrayList<HashMap<String, String>> studySets = runSqlQuery(
+          pool,
+          SEARCH_SQL_STATEMENT,
+          search_word
+        );
+        response.setContentType("application/json;");
+        Gson gson = new Gson();
+        String studySetsResult = gson.toJson(studySets);
+        response.getWriter().println(studySetsResult);
+      } catch (SQLException ex) {
+        throw new RuntimeException(
+          "There is an error with your sql statement ... ",
+          ex
+        );
+      }
+    } else {
```

Comment:
You could have a return; at the end of the if statement to avoid using an else statement so that everything is not being nested

Reply:
I like that idea